src/experiments.js

```diff
@@ -54,6 +57,112 @@ export function isCanary(win) {
   return !!(win.AMP_CONFIG && win.AMP_CONFIG.canary);
 }
 
+/**
+ * Determines if the specified experiment is on or off for origin trials.
+ * Callers shouldcheck if the experiment is already enabled before calling this
+ * function.
+ *
+ * The promise returned by this function will resolve to true IFF the specified
+ * experiment is enabled for this origin.
+ * The promise returned by this function will resolve to false if:
+ *   1. The experiment meta tag couldn't be found,
+ *   2. Crypto isn't available,
+ *   3. The experiment is not specified in the experiment token
+ *   4. The experiment has expired.
+ * The promise returned by this function will reject with an error if:
+ *   1. The experiment meta tag is present without a token
+ *   2. The token is malformed (e.g. non-existant version number)
+ *   3. The token is not for this origin
+ *   4. The experiments data was not signed with our private key
+ * @param {!Window} win
+ * @param {string} experimentId
+ * @param {!Object} opt_publicJwk Used for testing only.
+ * @return {!Promise<boolean>}
+ */
+export function isExperimentOnForOriginTrial(win, experimentId, opt_publicJwk) {
+  if (isExperimentOn(win, experimentId)) {
+    return Promise.resolve(true);
+  }
+  const crypto = cryptoFor(win);
+  const meta =
+      win.document.head.querySelector('meta[name="amp-experiment-token"]');
+  if (!meta || !crypto.isCryptoAvailable()) {
+    return Promise.resolve(false);
+  }
+  const token = meta.getAttribute('content');
+  if (!token) {
+    return Promise.reject(new Error('Unable to read experiments token'));
+  }
+  /**
+   * token = encode64(version + length + config + sign(config, private_key))
+   * version = 1 byte version of the token format (starting at 0x0)
+   */
+  let current = 0;
+  const bytes = stringToBytes(atob(token));
+  const version = bytes[current];
+  if (version !== 0) {
+    // Unrecognized version number
+    const error =
+        new Error(`Unrecognized experiments token version: ${version}`);
+    return Promise.reject(error);
+  }
+  current++;
+ /**
+  * Version 0:
+  * length = 4 bytes representing number of bytes in config
+  * config = string containing the experiment ID, origin URL, etc.
+  */
+  const bytesForConfigSize = 4;
+  const configSize =
```

Comment:
All of this byte array stuff is way more complicated than necessary. Why not just use a comma delimited string?

```js
${VERSION = 0},${BASE_64},${SIGNATURE}
```